src/main/java/at/hannibal2/skyhanni/features/misc/MagicalPowerDisplay.kt

```diff
@@ -0,0 +1,101 @@
+package at.hannibal2.skyhanni.features.misc
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.RenderItemTipEvent
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.RegexUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.RegexUtils.matches
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import io.github.moulberry.notenoughupdates.util.stripControlCodes
+import net.minecraft.item.ItemStack
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+@SkyHanniModule
+object MagicalPowerDisplay {
+    private val config get() = SkyHanniMod.feature.misc.mpDisplay.enabled
+    private val colored get() = SkyHanniMod.feature.misc.mpDisplay.colored
+    private fun isEnabled() = config && LorenzUtils.inSkyBlock
+
+    private val MPMap = mapOf(
```

Comment:
why create this when you can add the mp into the enum

Reply:
DIVINE, ULTIMATE and SUPREME does not have any MP data

Reply:
I'm pretty sure Supreme rarity doesn't even exist at all anymore.

Reply:
> DIVINE, ULTIMATE and SUPREME does not have any MP data

then map those to 0

Reply:
MP values are only used for this feature so changing it in the enum itself "poisons" the enum.
I don't think it's that necesary.

Reply:
I think if you don't want the pollute the enum it would be better to make this a function with a when statement so that the explicit not null is not needed on the getter of the map.

Reply:
I wrote this before seeing hanni say to remove the AccessoryRarity enum
So this review is obsolete yes

Reply:
Function > map imo too

Reply:
> Function > map imo too

Make sense. Will change later.